packages/driver/src/cy/keyboard.ts

```diff
@@ -65,7 +65,7 @@ const dateRe = /^\d{4}-\d{2}-\d{2}/
 const monthRe = /^\d{4}-(0\d|1[0-2])/
 const weekRe = /^\d{4}-W(0[1-9]|[1-4]\d|5[0-3])/
 const timeRe = /^([0-1]\d|2[0-3]):[0-5]\d(:[0-5]\d)?(\.[0-9]{1,3})?/
-const dateTimeRe = /^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}/
+const dateTimeRe = /^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}(:[0-9]{2})?(\.[0-9]{1,3})?/
```

Comment:
It's kind of odd we block keyboard input since of allowing users to ensure their input patterns are valid and handled as expected.

Reply:
I thought that was odd as well. I believe in this case it's because we don't actually simulate typing, we set the value for the provided datetime all at once (detecting the need to do this with [this helper here](https://github.com/cypress-io/cypress/blob/79f6651b1b5b93a88096c8d9d256a76c0b650528/packages/driver/src/dom/elements/elementHelpers.ts#L5). I guess this standardizes things since users don't ever type a raw date time, they use the browser UI to set individual parts of the date and time, or pick something from a calendar widget etc. It does seem a bit imprecise to use `type()` for this really.